docs/core/integration-quality-scale/rules/iqs027.md

```diff
@@ -0,0 +1,26 @@
+---
+title: "IQS027 - Dependency is async"
+---
+
+IQS027 is as follows:
+
+Dependency is async
+
+## Reasoning
+
+Home Assistant works with asyncio to be efficient.
+If your library is not async, it needs to utilize executor jobs to not block the event loop.
+Ideally, your library should be async.
```

Comment:
```suggestion
Home Assistant works with asyncio to be efficient when handling tasks.
To avoid switching context between the asyncio event loop and other threads, which is costly performance wise, ideally, your library should also use asyncio.
```